traffic_ops/traffic_ops_golang/profile/profiles.go

```diff
@@ -360,3 +362,294 @@ type) VALUES (
+			return
+		}
+		profileList = append(profileList, profile)
+	}
+	rows.Close()
+	profileInterfaces := []interface{}{}
+	for _, p := range profileList {
+		// Attach Parameters if the 'param' parameter is sent
+		if _, ok := inf.Params["param"]; ok {
+			p.Parameters, err = ReadParameters(inf.Tx, inf.User, &p.ID)
+			if err != nil {
+				api.HandleErr(w, r, tx.Tx, http.StatusInternalServerError, nil, fmt.Errorf("profile read: error reading parameters for a profile: %w", err))
+				return
+			}
+		}
+		profileInterfaces = append(profileInterfaces, p)
+	}
+
+	api.WriteResp(w, r, profileInterfaces)
+	return
+}
+
+// Create a Profile for APIv5
+func Create(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+	tx := inf.Tx.Tx
+
+	profile, readValErr := readAndValidateJsonStruct(r)
+	if readValErr != nil {
+		api.HandleErr(w, r, tx, http.StatusBadRequest, readValErr, nil)
+		return
+	}
+
+	//check if user can modify.
+	if len(strings.TrimSpace(profile.CDNName)) != 0 || profile.CDNID != 0 {
+		userErr, sysErr, statusCode := canProfileBeAlteredByCurrentUser(inf.User.UserName, inf.Tx.Tx, &profile.CDNName, &profile.CDNID)
+		if userErr != nil || sysErr != nil {
+			api.HandleErr(w, r, tx, statusCode, userErr, sysErr)
+			return
+		}
+	}
+
+	// check if profile already exists
+	var count int
+	err := tx.QueryRow("SELECT count(*) from profile where name=$1", profile.Name).Scan(&count)
+	if err != nil {
+		api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, fmt.Errorf("error: %w, when checking if profile '%s' exists", err, profile.Name))
+		return
+	}
+	if count == 1 {
+		api.HandleErr(w, r, tx, http.StatusBadRequest, fmt.Errorf("profile:'%s' already exists", profile.Name), nil)
+		return
+	}
+
+	// create profile
+	query := `INSERT INTO profile (name, cdn, type, routing_disabled, description) 
+	VALUES ($1, $2, $3, $4, $5) 
+	RETURNING id, last_updated, name, description, (select name FROM cdn where id = $2), cdn, routing_disabled, type`
+
+	err = tx.QueryRow(query, profile.Name, profile.CDNID, profile.Type, profile.RoutingDisabled, profile.Description).
+		Scan(&profile.ID, &profile.LastUpdated, &profile.Name, &profile.Description, &profile.CDNName, &profile.CDNID, &profile.RoutingDisabled, &profile.Type)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			api.HandleErr(w, r, tx, http.StatusInternalServerError, fmt.Errorf("error: %w in creating profile:%s", err, profile.Name), nil)
+			return
+		}
+		usrErr, sysErr, code := api.ParseDBError(err)
+		api.HandleErr(w, r, tx, code, usrErr, sysErr)
+		return
+	}
+
+	alerts := tc.CreateAlerts(tc.SuccessLevel, "profile was created.")
+	w.Header().Set("Location", fmt.Sprintf("/api/%d.%d/profiles?id=%d", inf.Version.Major, inf.Version.Minor, profile.ID))
```

Comment:
Please use [`github.com/apache/trafficcontrol/lib/go-rfc.Location`](https://pkg.go.dev/github.com/apache/trafficcontrol@v7.0.1+incompatible/lib/go-rfc#pkg-constants) instead of an inline string.